### DISCUSSION TITLE
Multicolored text

### DISCUSSION THREAD

**Discussion by TheLastochka** [AllenDang/giu#673] [Q&A]
I want to create a widget to display the log text. To do this, I chose the InputTextMultiline widget with the ReadOnly flag. But I'm facing the problem that I can't color the text in different colors. I want to color the error text red, the information text green, etc. I found a way to color the entire text in one color: using Style(). But how can I make it multicolored?

**Comment by gucio321** [comment]
Sadly it isn't possible for input text, but you can use ChildWidget with plain text that you can color with Style

**Reply by TheLastochka** [reply]
Can you give me an example?) I don't understand how to use ChildWidget.

**Comment by AllenDang** [comment]
I think you are trying to create some kind of syntax highlighter. There are two approaches to do that:
1. Read the source code of code editor widget, it's written in C++, and it has a syntax highlighter mechanism inside, should be easy to create a highlighter.
2. Use canvas,  set color to the keywords when drawing the string.
